src/Navbar.vue

```diff
@@ -49,6 +43,28 @@ export default {
   computed: {
     slots () {
       return this.$slots
+    },
+    themeOptions () {
+      // If the value of `type` is not primary | dark | light | none,
+      // then explicitly set isNavbarDark and isBgPrimary to true.
+      const isNavbarDark = (this.type === 'primary' || this.type === 'dark')
+        ? true
+        : (this.type === 'light' || this.type === 'none')
+          ? false
+          : true
+      const isBgPrimary = (this.type === 'primary')
+        ? true
+        : (this.type === 'dark' || this.type === 'light' || this.type === 'none')
+          ? false
+          : true
+
+      return {
+        'navbar-dark': isNavbarDark,
+        'navbar-light': (this.type === 'light'),
+        'bg-primary': isBgPrimary,
+        'bg-dark': (this.type === 'dark'),
+        'bg-light': (this.type === 'light')
+      }
```

Comment:
1. Prefer golden path rule whenever possible.
2. The equivalent of not using a valid theme keyword is the same as using the 'primary' theme.

```js
    themeOptions () {
      if (this.type === 'none') {
        return {};
      }
      
      const allowedThemes = ['primary', 'dark', 'light'];
      
      if (!allowedThemes.includes(this.type)) {
        this.type = 'primary';
      }
      
      return {
        'navbar-dark': (this.type === 'primary' || this.type === 'dark'),
        'navbar-light': (this.type === 'light'),
        'bg-primary': (this.type === 'primary'),
        'bg-dark': (this.type === 'dark'),
        'bg-light': (this.type === 'light')
      }
    } 
```

Reply:
Ah, thanks for this. Should have occurred to me that I can do an early return with `type === none`.

Reply:
A `switch` statement seems clearer.

```js
switch (this.type) {
  case 'dark':
    return 'navbar-dark bg-dark';
  ...
}
```

Reply:
> A `switch` statement seems clearer.
> 
> ```js
> switch (this.type) {
>   case 'dark':
>     return 'navbar-dark bg-dark';
>   ...
> }
> ```

Excellent point.

Reply:
Nice, I've pushed the changes 👍